src/main/java/org/openrewrite/java/dropwizard/annotation/AddClassAnnotation.java

```diff
@@ -0,0 +1,102 @@
+import static org.openrewrite.java.JavaParser.runtimeClasspath;
+
+public abstract class AddClassAnnotation extends Recipe {
+    private final String annotationText;
+    private final Boolean annotateSubclasses;
+
+    public AddClassAnnotation(String annotationText, Boolean annotateSubclasses) {
+        this.annotationText = annotationText;
+        this.annotateSubclasses = annotateSubclasses;
+    }
+
+    protected abstract boolean shouldAddAnnotation(ClassDeclaration cd);
+
+    @Override
+    public JavaIsoVisitor<ExecutionContext> getVisitor() {
+        return new JavaIsoVisitor<ExecutionContext>() {
+            @Override
+            public ClassDeclaration visitClassDeclaration(
+                    ClassDeclaration classDeclaration, ExecutionContext ctx) {
+                ClassDeclaration cd = super.visitClassDeclaration(classDeclaration, ctx);
+
+                // if annotation parameters are given as well
+                String annotationType = annotationText.split("[(<]")[0].trim();
+
+                if (service(AnnotationService.class)
+                        .matches(getCursor(), new AnnotationMatcher(annotationType))) {
+                    return cd;
+                }
+
+                boolean shouldAdd = shouldAddAnnotation(cd);
+
+                if (!shouldAdd && TRUE.equals(annotateSubclasses)) {
+                    shouldAdd = shouldAddAnnotationToAnyParentClass();
+                }
+
+                if (!shouldAdd) {
+                    return cd;
+                }
+
+                maybeAddImport(annotationType);
+
+                ClassDeclaration updated =
+                        JavaTemplate.builder("@#{}")
+                                .javaParser(fromJavaVersion().classpath(runtimeClasspath()))
+                                .imports(annotationType)
+                                .build()
+                                .apply(
+                                        updateCursor(cd),
+                                        cd.getCoordinates().addAnnotation(comparing(J.Annotation::getSimpleName)),
+                                        annotationText);
+                doAfterVisit(service(ImportService.class).shortenFullyQualifiedTypeReferencesIn(updated));
+                return maybeAutoFormat(cd, updated, ctx);
+            }
+
+            private boolean shouldAddAnnotationToAnyParentClass() {
+                try {
+                    getCursor().dropParentUntil(
+                                    x -> {
+                                        if (x instanceof J.ClassDeclaration) {
+                                            return shouldAddAnnotation((J.ClassDeclaration) x);
+                                        }
+                                        return false;
+                                    });
+
+                    return true;
+                } catch (IllegalStateException e) {
+                    return false;
+                }
```

Comment:
I'm not quite sure what's going on here; could you explain what you're trying to do? Then we can figure out if there's a better way than having this exception handling for regular flows.